plugins/mana/plugin.go

```diff
@@ -189,3 +179,77 @@ func pruneStorages() {
 //		storages[vectorType].Shutdown()
 //	}
 //}
+
+// GetHighestManaNodes returns the n highest type mana nodes in descending order.
+// It also updates the mana values for each node.
+func GetHighestManaNodes(manaType mana.Type, n int) []mana.NodeIDMana {
+	bmv := baseManaVectors[manaType]
+	return bmv.GetHighestManaNodes(n)
+}
+
+// GetManaMap return type mana perception of the node.
+func GetManaMap(manaType mana.Type) mana.Map {
+	return baseManaVectors[manaType].GetManaMap()
+}
+
+// GetAccessMana returns the access mana of the node specified.
+func GetAccessMana(nodeID identity.ID) (float64, error) {
+	return baseManaVectors[mana.AccessMana].GetMana(nodeID)
+}
+
+// GetConsensusMana returns the consensus mana of the node specified.
+func GetConsensusMana(nodeID identity.ID) (float64, error) {
+	return baseManaVectors[mana.ConsensusMana].GetMana(nodeID)
+}
+
+// GetNeighborsMana returns the type mana of the nodes neighbors
+func GetNeighborsMana(manaType mana.Type) (mana.Map, error) {
+	neighbors := gossip.Manager().AllNeighbors()
+	res := make(mana.Map)
+	for _, n := range neighbors {
+		value, err := baseManaVectors[manaType].GetMana(n.ID())
+		if err != nil {
+			return nil, err
+		}
+		res[n.ID()] = value
+	}
+	return res, nil
+}
+
+// GetAllManaMaps returns the full mana maps for comparison with the perception of other nodes.
+func GetAllManaMaps() map[mana.Type]mana.Map {
+	res := make(map[mana.Type]mana.Map)
+	for manaType := range baseManaVectors {
+		res[manaType] = GetManaMap(manaType)
+	}
+	return res
+}
+
+// OverrideMana sets the nodes mana to a specific value.
+// It can be useful for debugging, setting faucet mana, initialization, etc.. Triggers ManaUpdated
+func OverrideMana(manaType mana.Type, nodeID identity.ID, bm *mana.BaseMana) {
+	baseManaVectors[manaType].SetMana(nodeID, bm)
+}
+
+//GetWeightedRandomNodes returns a weighted random selection of n nodes.
+func GetWeightedRandomNodes(n int, manaType mana.Type) mana.Map {
+	rand.Seed(time.Now().UTC().UnixNano())
+	manaMap := GetManaMap(manaType)
+	var choices []mana.RandChoice
+	for nodeID, manaValue := range manaMap {
+		choices = append(choices, mana.RandChoice{
+			Item:   nodeID,
+			Weight: int(manaValue * 1000000), //scale float mana to int
```

Comment:
in the current implementation, transferring 1 IOTA results in getting 1 mana. Therefore, I don't think we need to multiply by that much.
I was thinking about just ignoring any node that has less, than let's say 0.001 mana, since we don't want our mana map to be bigger than necessary. (and due to the asymptotic decay of mana 2, it will never be exactly zero in theory..)
Also, if 1 IOTA is the smallest amount, 1 mana should represent quite an insignificant weight in the network.

Reply:
ok. I'll use 1000 for now. We can change this as we see fit.